### PR TITLE
Removes automappable constants, manual maps enum constants were auto mapping fails

### PR DESCRIPTION
**Pull request by NeunEinser**
There are a whole bunch of unnecessary mappings in this project.
This is an attempt to remove most of them.

Mostly this removes enum constant mappings. It also adds them, where sting fails to provide the correct auto mappings. All mapped enum constants in this pr are derived directly from the byte code.
In some cases, a rename of enum classes was needed because of slightly changed meaning of enum constants.
This also removes other mappings that are properly auto mapped and sanitizes some of them.

Here is a list of enum classes, were at least partial mappings are necessary, because they either do not get auto mapped at all, or are automapped based on a string value inside of the enum constructor, rather than the actual constant name.
```
No automapping possible:
com.mojang.blaze3d.platform.GlStateManager.RenderMode
net.minecraft.enchantment.EnchantmentTarget
net.minecraft.entity.ai.brain.task.CompositeTask.RunMode
net.minecraft.entity.boss.dragon.EnderDragonSpawnState
net.minecraft.network.NetworkState
net.minecraft.util.SystemUtil.OperatingSystem
net.minecraft.util.math.AxisCycleDirection
net.minecraft.util.math.Direction.Axis
net.minecraft.world.biome.layer.ScaleLayer
net.minecraft.world.chunk.UpgradeData.b

Wrong automapping:
net.minecraft.block.BlockRenderLayer
net.minecraft.block.entity.BannerPattern
net.minecraft.client.audio.Sound.SoundEvent
net.minecraft.client.gl.GlShader.Type
net.minecraft.client.options.AoOption
net.minecraft.client.options.AttackIndicator
net.minecraft.client.options.ChatVisibility
net.minecraft.client.options.CloudRenderMode
net.minecraft.client.options.NarratorOption
net.minecraft.client.options.ParticlesOption
net.minecraft.client.render.VertexFormatElement.Type
net.minecraft.client.render.VertexFormatElement.Format
net.minecraft.client.util.InputUtil.Type
net.minecraft.item.ArmorMaterials
net.minecraft.resource.ResourcePackCompatibility
net.minecraft.resource.ResourceType
net.minecraft.sound.SoundCategory
net.minecraft.util.math.Direction.AxisDirection
net.minecraft.world.GameMode
net.minecraft.world.biome.Biome.Category
```

### CODE REVIEW

File: mappings/net/minecraft/world/biome/Biome.mapping [hunk 1]
```diff
@@ -23,33 +23,13 @@ CLASS bie net/minecraft/world/biome/Biome
...
-		FIELD i THE_END Lbie$b;
-		FIELD j BEACH Lbie$b;
-		FIELD k FOREST Lbie$b;
-		FIELD l OCEAN Lbie$b;
-		FIELD m DESERT Lbie$b;
-		FIELD n RIVER Lbie$b;
-		FIELD o SWAMP Lbie$b;
-		FIELD p MUSHROOM Lbie$b;
-		FIELD q NETHER Lbie$b;
+		FIELD i THEEND Lbie$b;
```
Comment (NeunEinser) [inline mappings/net/minecraft/world/biome/Biome.mapping:26]: This is not a typo (by me). The constant appears like that without an underscore in the byte code. Might be a typo by Mojang, though.

**Review by asiekierka — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Looks good, do we want to merge in this now?

**Comment by modmuss50** [comment]
merge conflicts need fixing, let me know if you want me to sort them out tomorrow.

**Review by Runemoro — APPROVED** [review]

**Comment by liach** [comment]
Are these constants automapped in jars produced by the gradle tasks? Back in pre-5 automapped constants are in intermediary names in the yarn name merged jar.

**Comment by modmuss50** [comment]
The gradle tasks do not populate these fields, open an issue about that and it can be added.

**Comment by NeunEinser** [comment]
Weird, my git tool didn't have anything here that could not be merged automatically.

**Comment by magneticflux-** [comment]
@modmuss50 I'm getting the intermediary names in the `Direction` enum. Is this the issue I should be watching? https://github.com/FabricMC/fabric-loom/issues/91

**Comment by modmuss50** [comment]
Im not 100% sure what your issue is, but you should be able to see enum names in your IDE. That issue you linked is something else unrelated to the code you see in your ide.

**Comment by magneticflux-** [comment]
I just updated to newer mappings (`1.14.2 Pre-Release 4+build.1`) and my Direction enum references broke. After changing them to the intermediary names in my code I can execute `runClient`, but the enum still has the intermediary names in `minecraft-1.14.2 Pre-Release 4-mapped-1.jar`.

**Comment by Chocohead** [comment]
Which Loom version are you running? Could theoretically be an outdated Stitch (although I don't think anything has changed in it particularly recently)

**Comment by magneticflux-** [comment]
I'm using `Fabric Loom: 0.2.2-SNAPSHOT Build(jenkins #11)`, which should be the most recent since I just ran `gradlew build --refresh-dependencies`.

**Comment by liach** [comment]
no, yarn is not related to loom. @magneticflux- please move this talk over to loom.

**Comment by magneticflux-** [comment]
My issue was resolved by updating to the latest 1.14.2 versions. I don't know what else changed.

**Pull request merged**